<launch>

  <node pkg="map_server" type="map_server" name="map_server_node" args="$(find bullproof_perception)/maps/robohouse.yaml">
    <remap from="map" to="map_test"/>
    <remap from="map_metadata" to="map_test_metadata"/>
    <param name="frame_id" value="map_test" />
  </node>

  <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="launch_prefix" default="" />

  <!-- configure camera input -->
  <arg name="camera_name" default="/webcam" />
  <arg name="image_topic" default="image_sync" />
  <arg name="queue_size" default="1" />

  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
    <param name="queue_size" type="int" value="$(arg queue_size)" />

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find bullproof_perception)/config/settings.yaml"/>
    <rosparam command="load" file="$(find bullproof_perception)/config/tags.yaml"/>
  </node>

  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/webcam/image_raw raw out:=/webcam/image_sync" />

  <node pkg="bullproof_perception" name="get_detections" type="detection.py"/>

  <!-- transforms -->
  <node pkg="tf" type="static_transform_publisher" name="tf_camera_map" args="0.0 0.0 0 0 0 0 head_camera map_test 20"/>
  <node pkg="bullproof_perception" type="localizer" name="localizer"/>


</launch>

